### DISCUSSION TITLE
Is is possible to use openpnp / opencv

### DISCUSSION THREAD

**Discussion by git9999999** [dcm4che/dcm4che#1107] [Q&A] [answered]
Hi 
I need to convert all sorts of Dicom Pictures to JPEG. I found example how to do this with dcom4che :-). As fare as i understand i need a native lib called opencv (https://opencv.org/). As this is not perfect for a Java project working with Maven. I wondered if i can replace this native lib with openpnp / opencv (https://github.com/openpnp/opencv)

For me it does not look like it is possible. The only way that maybe could work would be to fork org.weasis.core:weasis-core-img:4.4.0 and replace the loader

```
public class OpenCVNativeLoader implements OpenCVInterface {
    public OpenCVNativeLoader() {
    }

    public void init() {
        System.loadLibrary(Core.NATIVE_LIBRARY_NAME);
        Logger.getLogger("org.opencv.osgi").log(Level.INFO, "Successfully loaded OpenCV native library.");
    }
}
```

with the one from openpnp

Do you have any Ideas to replace the native libs from opencv?
Regards T

**Comment by nroduit** [comment, marked as answer]
You cannot use another library because the original one contains the DICOM implementation for compression and decompression.

You should check this [library](https://github.com/nroduit/weasis-dicom-tools/tree/img) if you want to have Java API for converting to a standard image format. You can get 5.25.1-IMG-RELEASE from t[his maven repository](https://github.com/nroduit/mvn-repo).

The API you need is [dcm2image](https://github.com/nroduit/weasis-dicom-tools/blob/img/src/main/java/org/dcm4che3/img/Transcoder.java#L80-L115). If you want to convert in jpeg, you should understand which Window/Level to apply, which mode pixel depth preserving or not...

Here are the features of dcm2image:
- Convert in several images formats (JPEG, PNG, TIF, JP2, PNM, BMP or HDR) 
- Option to preserve the pixel depth with some formats (eg. 16-bit TIF, double values in TIF or HDR)
- Capabilities to apply Modality, VOI and Presentation LUT with 8-bit output images
- Support multiframe and/or multi-fragments. For multiframe output images are created with an index.
- Apply a Presentation State (W/L, LUTs, overlay).
- Set the color of overlays in image or Presentation State

**Reply by git9999999** [reply]
Hi nroduit

Thanks for the answer :-), i was expecting this :-(. But i am happy to be sure now. :-) 

I already adapted the [dcm2image](https://github.com/nroduit/weasis-dicom-tools/blob/img/src/main/java/org/dcm4che3/img/Transcoder.java#L80-L115) to my needs.

Regards G

**Reply by nroduit** [reply]
> I already adapted the [dcm2image](https://github.com/nroduit/weasis-dicom-tools/blob/img/src/main/java/org/dcm4che3/img/Transcoder.java#L80-L115) to my needs.

If there are things missing in the API please report them or propose a modification by a Pull Request.
